app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = Settings.valid_email_regex
+  before_save { email.downcase! }
```

Comment:
downcase khác gì downcase! không e ?

Reply:
+ downcase và downcase! dùng để in thường các ký tự trong chuỗi
 + downcase! có thể thay đổi giá trị của chuỗi ban đầu còn downcase thì không,
downcase! sẽ trả về nil nếu xâu ký tự không có gì thay đổi